### PR TITLE
Add -share-server flag for sharing integration test resources

### LINKED ISSUE #3442: Add option to share listener/handler/backend between integration tests
### What should be done?

Currently, our integration tests share a few things:
* debug handler with pprof, execution tracer, metrics, etc;
* `listenerMetrics` for the above handler;
* OTLP exporter for traces.

There should be a flag `-share-server` (false by default) that would make all tests share a single instance of listener/handler/backend. It should be started once in `setup.Startup` and stopped in `setup.Shutdown`. Functions like `setupListener` should return that shared listener/handler/backend.

This is likely the underlying reason for #3409, but fixing it or other integration test failures is not in the scope.
`Taskfile.yml` changes are not in the scope. That flag should be tested manually.

The MongoDB client should not be shared.

### Where?

https://github.com/FerretDB/FerretDB/tree/main/integration/setup

### Definition of Done

- flag implemented;
- spot refactorings done.

### PR DESCRIPTION
**Pull request by Mihai22125**
# Description

Closes #3442 .

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [ ] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [ ] I marked all done items in this checklist.

### CODE REVIEW

File: integration/setup/setup.go [hunk 1]
```diff
@@ -65,6 +65,8 @@ var (
 
 	useNewPgF   = flag.Bool("use-new-pg", false, "use new PostgreSQL backend")
 	useNewHanaF = flag.Bool("use-new-hana", false, "use new SAP HANA backend")
+
+	shareServerF = flag.Bool("share-server", true, "make all tests share a single instance of listener/handler/backend")
```
Comment (AlekSi) [inline integration/setup/setup.go:69]: We should switch to `false` by default when this PR is ready

File: integration/setup/listener.go [hunk 2]
```diff
@@ -101,71 +83,157 @@ func setupListener(tb testtb.TB, ctx context.Context, logger *zap.Logger) string
...
+	}
+
+	uri := listenerMongoDBURI(tb, hostPort, unixSocketPath, tlsAndAuth)
+
+	logger.Info("Listener started", zap.String("handler", handler), zap.String("uri", uri))
+
+	return uri
+}
+
+func initListener(ctx context.Context) (string, *clientconn.Listener) {
```
Comment (AlekSi) [inline integration/setup/listener.go:134]: I think that function could accept `fatalf func(format string, args ...any)` that could be either `tb.Fatalf` or `zap.S().Fatalf`

**Review by AlekSi — COMMENTED** [review]
Integration tests fail

**Comment by AlekSi** [comment]
@Mihai22125 any news? Anything we could help you with?

**Comment by Mihai22125** [comment]
@AlekSi, I found one of the issues to be that my changes were deleting some files too early. Unfortunately, I've had some busy days, but I will address it in the next few days.

**Comment by AlekSi** [comment]
https://github.com/FerretDB/FerretDB/issues/3442#issuecomment-1758206809 Sorry once again

**Pull request closed without merging**